src/libtest/lib.rs

```diff
@@ -923,19 +830,43 @@ pub fn run_tests_console(opts: &TestOpts, tests: Vec<TestDescAndFn>) -> io::Resu
         }
     }
 
-    let mut st = ConsoleTestState::new(opts, None::<io::Stdout>)?;
+    let output = match term::stdout() {
+        None => Raw(io::stdout()),
+        Some(t) => Pretty(t),
+    };
+
+    let max_name_len = if let Some(t) = tests.iter().max_by_key(|t| len_if_padded(*t)) {
+        let n = t.desc.name.as_slice();
+        n.len()
+    }
+    else {
```

Comment:
Such else formatting looks unconventional. We can try to eliminate it altogether with `.map(|t| t.desc.name.as_slice().len()).unwrap_or(0)`